File: 2array/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
// Get input from user
    string textInput = get_string("Text: ");
    int countLett = 0, countWord = 1, countSent = 0;
    

//Objects counting cases (letters, words and sentences)
    for (int i = 0, n = strlen(textInput); n > i; i++)
    {
        if (isalpha(textInput[i]))
        {
            countLett++;
        }
        if ((isspace(textInput[i])) || (isblank(textInput[i]))) 
        {
            countWord++;
        }
        if (textInput[i] == '.' || textInput[i] == '!' || textInput[i] == '?')
        {
            countSent++;
        }
    }

//Coleman-Liau test aplication
    float L = (float)countLett / (float)countWord * 100, S = (float)countSent / (float)countWord * 100;
    float grade = 0.0588 * L - 0.296 * S - 15.8;
    
//Conditional of grade's level
    if (round(grade) >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (round(grade) < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }

}

File: 6python/pset6/readability.py
from cs50 import get_string

text = get_string("Text: ")
countLetters, countWords, countSentences = 0, 1, 0

for i in text:
    if (i.isalpha()):
        countLetters += 1
    elif (i == ' '):
        countWords += 1
    elif (i == '.' or i == '!' or i == '?'):
        countSentences += 1

def get_index(str):
    L = countLetters / countWords * 100
    S = countSentences / countWords * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)

grade = get_index(text)
if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before grade 1")
else:
    print("Grade", grade)